src/contexts/NftRewards/NftRewardsProvider.tsx

```diff
@@ -20,8 +19,12 @@ export const NftRewardsProvider: React.FC = ({ children }) => {
   const { projectMetadata, projectId } = useContext(ProjectMetadataContext)
 
   const dataSourceAddress = fundingCycleMetadata?.dataSource
+  const contractVersion = useJB721DelegateVersion({
+    dataSourceAddress,
+  })
+
   // don't fetch stuff if there's no datasource in the first place.
-  const { value: hasNftRewards } = useHasNftRewards()
+  const hasNftRewards = Boolean(contractVersion)
```

Comment:
Now we only call `useJB721DelegateVersion` once per pageload. hasNftRewards will be `false` if no contractVersion.